pydicom/pixel_data_handlers/numpy_handler.py

```diff
@@ -134,8 +127,37 @@ def get_pixeldata(dicom_dataset):
             "Amount of pixel data %d does not "
             "match the expected data %d" %
             (length_of_pixel_array, padded_length))
-    if expected_length != padded_length:
+
+    # the checks above confirmed the amount of data in PixelData is what
+    #  we expect, now we can unpack it
+    if dicom_dataset.BitsAllocated == 1:
+        # if single bits are used for binary representation, a uint8 array
+        # has to be converted to a binary-valued array (that is 8 times bigger)
+        bit = 0
+        pixel_array = \
+            numpy.ndarray(shape=(length_of_pixel_array*8), dtype='uint8')
+        # bit-packed pixels are packed from the right; i.e., the first pixel
+        #  in the image frame corresponds to the first from the right bit of
+        #  the first byte of the packed PixelData!
+        #  See the following for details:
+        #  * DICOM 3.5 Sect 8.1.1 (explanation of bit ordering)
+        #  * DICOM Annex D (examples of encoding)
+        print("Type: "+str(type(pixel_bytearray[0])))
+        for byte in pixel_bytearray:
+            if isinstance(byte, str):
+                byte = ord(byte)
+            for bit in range(bit, bit+8):
+                pixel_array[bit] = byte & 1
+                byte >>= 1
```

Comment:
On a second thought - it might not be a good idea to use `isinstance()` in the loop for performance reasons. Better use `if in_py2`.

Reply:
How about thinking along the lines of:
```
numpy.flip(
    numpy.unpackbits(pixel_bytearray).reshape((length_of_pixel_array, 8)), 
    axis=1).reshape(length_of_pixel_array * 8)

```
I am pretty sure something like this should work....

Reply:
@rhaxton can you make a PR with your proposed change to my branch https://github.com/qiicr/pydicom/tree/fix-seg-pixeldata? To me personally, the code I wrote is more readable, and I am not sure if the suggested change leads to any performance improvements (which also I am not sure are critical in this situation). I am not an expert in numpy, so I would defer to someone else doing the numpy-specific optimizations.

Reply:
Actually, I take it back - I oppose this suggestion of changing to use `unpackbits`, since it means we would only be able to unpack in python 3. Not clear at all what is the advantage of restricting this operation to python 3 only.

Reply:
> Better use if `in_py2`

Done